src/document.jl

```diff
@@ -84,42 +84,38 @@ end
 
 #### parse and free
 
-function parse_file(filename::AbstractString)
-    p = ccall((:xmlParseFile,libxml2), Xptr, (Cstring,), filename)
+function _check_result(p)
     p != C_NULL || throw(XMLParseError("Failure in parsing an XML file."))
     XMLDocument(p)
 end
 
-function parse_file(filename::AbstractString, encoding, options::Integer)
-    p = ccall((:xmlReadFile,libxml2), Xptr, (Cstring, Ptr{Cchar}, Cint),
-        filename, encoding, options)
-    p != C_NULL || throw(XMLParseError("Failure in parsing an XML file."))
-    XMLDocument(p)
-end
+parse_file(filename::AbstractString) =
+    _check_result(ccall((:xmlParseFile,libxml2), Xptr, (Cstring,), filename))
 
-function parse_string(s::AbstractString)
-    p = ccall((:xmlParseMemory,libxml2), Xptr, (Xstr, Cint), s, sizeof(s) + 1)
-    p != C_NULL || throw(XMLParseError("Failure in parsing an XML string."))
-    XMLDocument(p)
-end
+parse_file(filename::AbstractString, encoding, options::Integer) =
+    _check_result(ccall((:xmlReadFile,libxml2), Xptr, (Cstring, Ptr{Cchar}, Cint),
+        filename, encoding, options))
+
+parse_string(s::AbstractString) =
+    _check_result(ccall((:xmlParseMemory,libxml2), Xptr, (Xstr, Cint), s, sizeof(s) + 1))
 
 
 #### output
 
 function save_file(xdoc::XMLDocument, filename::AbstractString; encoding::AbstractString="utf-8")
-    ret = ccall((:xmlSaveFormatFileEnc,libxml2), Cint, (Cstring, Xptr, Cstring, Cint),
-        filename, xdoc.ptr, encoding, 1)
-    if ret < 0
-        throw(XMLWriteError("Failed to save XML to file $filename"))
-    end
-    return @compat Int(ret)  # number of bytes written
+    ret = ccall((:xmlSaveFormatFileEnc,libxml2), Cint,
+                (Cstring, Xptr, Cstring, Cint),
+                filename, xdoc.ptr, encoding, 1)
+    ret < 0 && throw(XMLWriteError("Failed to save XML to file $filename"))
+    Int(ret)  # number of bytes written
 end
 
 function Base.string(xdoc::XMLDocument; encoding::AbstractString="utf-8")
-    buf_out = Vector{Xstr}(1)
-    len_out = Vector{Cint}(1)
-    ccall((:xmlDocDumpFormatMemoryEnc,libxml2), Void, (Xptr, Ptr{Xstr}, Ptr{Cint}, Cstring, Cint),
-        xdoc.ptr, buf_out, len_out, encoding, 1)
+    buf_out = @compat Vector{Xstr}(uninitialized, 1)
+    len_out = @compat Vector{Cint}(uninitialized, 1)
```

Comment:
The `@compat` shouldn't be necessary

Reply:
I don't recall any personal code style choices, I did split a few lines that didn't follow Julia's own guidelines on submissions: "i.e. try to adhere to a 92 character line length limit"

Reply:
Those changes are good. I was mostly referring to: 
- [Adding extra spaces](https://github.com/JuliaIO/LightXML.jl/pull/78/files#diff-2202e1de355727f6f7ebd5cf4e7e8c27R45)
- [Switching from `function` to `=` definitions](https://github.com/JuliaIO/LightXML.jl/pull/78/files#diff-2202e1de355727f6f7ebd5cf4e7e8c27R238)
- [Updating `if` blocks](https://github.com/JuliaIO/LightXML.jl/pull/78/files#diff-2202e1de355727f6f7ebd5cf4e7e8c27R275)